app/components/preprint-file-renderer.js

```diff
@@ -0,0 +1,91 @@
+import Ember from 'ember';
+import Component from '@ember/component';
+import DS from 'ember-data';
+import { computed } from '@ember/object';
+import { inject as service } from '@ember/service';
+import loadAll from 'ember-osf/utils/load-relationship';
+import Analytics from 'ember-osf/mixins/analytics';
+import fileDownloadPath from '../utils/file-download-path';
+
+/**
+ * @module ember-preprints
+ * @submodule components
+ */
+
+/**
+ * Displays primary file and allows downloading of versions
+ *
+ * Sample usage:
+ * ```handlebars
+ * {{preprint-file-renderer
+ *      preprint=model
+ *      provider=model.provider
+ *      primaryFile=primaryFile
+ * }}
+ * ```
+ * @class preprint-file-renderer
+ */
+
+const { A } = Ember;
+const { PromiseArray } = DS;
+
+export default Component.extend(Analytics, {
```

Comment:
This is similar to the supplementary-file-browser, except the supplementary files bit is gone.  This just loads the primary file and allows downloading of versions.